apis/fluentbit/v1alpha2/plugins/filter/log_to_metrics_types.go

```diff
@@ -0,0 +1,112 @@
+package filter
+
+import (
+	"fmt"
+	"github.com/fluent/fluent-operator/v2/apis/fluentbit/v1alpha2/plugins"
+	"github.com/fluent/fluent-operator/v2/apis/fluentbit/v1alpha2/plugins/params"
+)
+
+// +kubebuilder:object:generate:=true
+
+// The Log To Metrics Filter plugin allows you to generate log-derived metrics. <br />
+// **For full documentation, refer to https://docs.fluentbit.io/manual/pipeline/filters/log_to_metrics**
+type LogToMetrics struct {
+	plugins.CommonParams `json:",inline"`
+	// Defines the tag for the generated metrics record
+	Tag string `json:"tag,omitempty"`
+	// Optional filter for records in which the content of KEY matches the regular expression.
+	// Value Format: FIELD REGEX
+	Regex []string `json:"regex,omitempty"`
+	// Optional filter for records in which the content of KEY does not matches the regular expression.
+	// Value Format: FIELD REGEX
+	Exclude []string `json:"exclude,omitempty"`
+	// Defines the mode for the metric. Valid values are [counter, gauge or histogram]
+	MetricMode string `json:"metric_mode,omitempty"`
+	// Sets the name of the metric.
+	MetricName string `json:"metric_name,omitempty"`
+	// Namespace of the metric
+	MetricNamespace string `json:"metric_namespace,omitempty"`
```

Comment:
all the json field should be `camelCase`:
metric_namespace -> metricNamespace
metric_subsystem -> metricSubsystem

etc.